### DISCUSSION TITLE
The priority of the fonts that Typst CLI chooses to use?

### DISCUSSION THREAD

**Discussion by Complex2-Liu** [typst/typst#3100] [Q&A] [answered]
According to [Arch Wiki][arch], Linux has **system-level** fonts path `/usr/share/fonts` or `/usr/local/share/fonts` and **user-level** fonts path `$XDG_DATA_HOME/fonts` or `$HOME/.fonts` (deprecated).

Typst CLI also provides a environment variable `TYPST_FONT_PATHS` (assume to be `$HOME/typst/fonts`), call it **typst-level** fonts path.

Typst also internally ships with some fonts, say New Computer Modern (NCM, file `ncm-internal.otf`), call it **internal-level** fonts path.

Suppose now I were not satisfied with the calligraphic font style of NCM, I slightly modified it through [fontforge] and generated 3 mutually different fonts: `ncm-system.otf`, `ncm-user.otf` and `ncm-typst.otf` and put them into the corresponding fonts path.

Although we have 4 different fonts, By `typst fonts | grep -i "Modern"` we see that typst detect only one of these fonts. So which one is of the highest priority? The typst-level one (`$HOME/typst/fonts/ncm-typst.otf`) or the internal-level one?

[arch]: https://wiki.archlinux.org/title/fonts#Manual_installation
[fontforge]: https://fontforge.org/en-US/

**Comment by laurmaedje** [comment, marked as answer]
Priority from lowest to highest: Internal, System, Typst-level.

**Reply by Complex2-Liu** [reply]
Silly me, I actually thought that Internal had the highest priority 😅.

**Reply by laurmaedje** [reply]
The priority is chosen such that it is possible to override.
